### ISSUE TITLE
Create directory on system without decoding encoded values in Slug 

### ISSUE THREAD

**Issue by csarven** [solid/node-solid-server#422]
Perhaps the title can be better with less assumptions as to what may be happening. Here is the code that I've used:

``` js
var options = {};
var url = 'https://example.org/';
var contentType = 'text/turtle';
var links = '<http://www.w3.org/ns/ldp#Container>; rel="type"';
var slug = encodeURIComponent('http://csarven.ca/foo');
var data = '';

var http = new XMLHttpRequest();
http.open('POST', url);
http.setRequestHeader('Content-Type', contentType);
http.setRequestHeader('Link', links);
if (slug && slug.length > 0) {
    http.setRequestHeader('Slug', slug);
}
if (!options.noCredentials) {
    http.withCredentials = true;
}
http.onreadystatechange = function() {
    if (this.readyState == this.DONE) {
        console.log(this);
    }
};
http.send(data);
```

Response has `Location: /http%3A%2F%2Fcsarven.ca%2Ffoo`.

What's physically created on the system is directory `http:` with subdirectory `csarven.ca` with file `foo`

Expected directory: `http%3A%2F%2Fcsarven.ca%2Ffoo`

This should be analogous to `mkdir "http%3A%2F%2Fcsarven.ca%2Ffoo"`

**Comment by csarven** [comment]
Oh there is more:

If I create directory `http%3A%2F%2Fcsarven.ca%2Ffoo` manually on the server, and then later POST to that directory, the POST goes through (i.e., it creates say resource `https://example.org/http%3A%2F%2Fcsarven.ca%2Ffoo/bar`. However, when I GET that container (`http://example.org/http%3A%2F%2Fcsarven.ca%2Ffoo`), it has `ldp:contains <http://example.org/bar>`. Expected: `ldp:contains <http://example.org/http%3A%2F%2Fcsarven.ca%2Ffoo/bar>`

**Comment by dmitrizagidulin** [comment]
Thanks, Sarven, that's helpful.

**Comment by dmitrizagidulin** [comment]
Fixed in PR #469, closing.
